.github/workflows/regression_tests.yml

```diff
@@ -0,0 +1,60 @@
+name: Regression Tests
```

Comment:
I tested this out in a seperate repo, but I cannot for the life of me get the commenting to work.
The basic idea is:
1. run regression tests -> output to file `regression_test_results.json`
2. get the diff between `regression_test_results.json` and `main:regression_test_results.json`
3. comment on PR with sth. along the lines of "test x is slower than in main"
4. ammend last commit, such that when PR gets merge `regression_test_results.json` becomes new baseline.

We could also do without the ammend, then we would have to maintain the baseline file manually. This might be preferred actually since it gives us full control over the baseline.